src/main/java/org/xmldb/api/security/AclEntry.java

```diff
@@ -185,6 +185,39 @@ public Builder setPermissions(AclEntryPermission... perms) {
       return this;
     }
 
+    /**
+     * Sets the permissions component of this builder. On return, the permissions component of this
+     * builder is a copy of the permissions in the given array.
+     *
+     * @param modeStr the permissions mode string
+     * @return this builder
+     */
+    public Builder setPermissions(String modeStr) {
```

Comment:
I am not sure this works for the string "r-x"

Reply:
I will also add that one too...